src/compiler/transpile.ts

```diff
@@ -72,8 +74,9 @@ export const transpileSync = (code: string, opts: TranspileOptions = {}): Transp
   try {
     if (shouldTranspileModule(results.inputFileExtension)) {
       const { config, compileOpts, transformOpts } = getTranspileConfig(opts);
-      patchTypescript(config, null);
-      transpileCode(config, compileOpts, transformOpts, results);
+      const validatedConfig = validateConfig(config, {}).config;
+      patchTypescript(validatedConfig, null);
+      transpileCode(validatedConfig, compileOpts, transformOpts, results);
```

Comment:
I'm hesitant we can safely make this change without it being potentially breaking, since this function is a part of our public Node API + used by Jest testing of Stencil projects. By creating a `ValidatedConfig`, we end up providing a config object with additional properties on it that folks may be (un)intentionally not providing/relying upon.